File: spec/java/src/io/kaitai/struct/spec/TestNonStandard.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.NonStandard;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestNonStandard extends CommonSpec {
    @Test
    public void testNonStandard() throws Exception {
        NonStandard r = NonStandard.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.foo(), 0x50);
    }
}

File: spec/lua/test_non_standard.lua
local luaunit = require("luaunit")

require("non_standard")

TestNonStandard = {}

function TestNonStandard:test_non_standard()
    local r = NonStandard:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.foo, 0x50)
end
